src/sched/lthread.c

```diff
@@ -580,13 +588,13 @@ int _lthread_sched_init(size_t stack_size)
 
     sched_stack_size = stack_size ? stack_size : MAX_STACK_SIZE;
 
-    struct schedctx* c = __scheduler_self();
+    struct lthread_sched* sched = lthread_get_sched();
 
-    c->sched.stack_size = sched_stack_size;
+    sched->stack_size = sched_stack_size;
 
-    c->sched.default_timeout = 3000000u;
+    sched->default_timeout = 3000000u;
 
-    memset(&c->sched.ctx, 0, sizeof(struct cpu_ctx));
+    memset(&sched->ctx, 0, sizeof(struct cpu_ctx));
```

Comment:
Not very important, but I wonder if this is actually necessary.  This field should be fully set by `_switch` the first time this scheduler runs an lthread and the zero value should never be read.